Tasks/AzureIoTEdgeV2/deployimage.ts

```diff
@@ -43,18 +43,42 @@ class azureclitask {
 
       this.loginAzure();
 
-      tl.debug('OS release:' + os.release());
-
+      tl.debug('OS release:' + os.release());      
       let showIotExtensionCommand = ["extension", "show", "--name", "azure-iot"];
       let result = tl.execSync('az', showIotExtensionCommand, Constants.execSyncSilentOption);
       if (result.code !== 0) { // The extension is not installed
+        let checkAzureIoTVersionExtensionCommand = ["--version"]; // temporary solution until the paho-mqtt situation goes away 
```

Comment:
>paho-mqtt situation goes away [](start = 98, length = 29)

Could you expand on the root cause and the work around?

Reply:
The root cause is that sometimes the hosted agent does not recognize that the azure-iot extension is installed so the temporary workaround is to verify the version that is currently installed for sanity check. It doesn't actually solve the issue within the task, only works as a devops script.